substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs

```diff
@@ -0,0 +1,263 @@
+            Ok(country_name) => country_name,
+            Err(e) => {
+                info!(
+                    "failed to parse country name for entity: {:?}, error: {:?}",
+                    k, e
+                );
+                let default_country_name_input: CountryNameInput =
+                    BoundedVec::try_from(node::DEFAULT_COUNTRY_NAME.to_vec()).unwrap();
+                info!("set default country name for entity");
+                <T as Config>::CountryName::try_from(default_country_name_input).unwrap()
+            }
+        };
+
+        let city_name_input: CityNameInput = BoundedVec::try_from(entity.city).unwrap();
+        let city = match <T as Config>::CityName::try_from(city_name_input) {
+            Ok(city_name) => city_name,
+            Err(e) => {
+                info!(
+                    "failed to parse city name for entity: {:?}, error: {:?}",
+                    k, e
+                );
+                let default_city_name_input: CityNameInput =
+                    BoundedVec::try_from(node::DEFAULT_CITY_NAME.to_vec()).unwrap();
+
+                info!("set default city name for entity");
+                <T as Config>::CityName::try_from(default_city_name_input).unwrap()
+            }
+        };
+
+        let new_entity = TfgridEntity::<T> {
+            version: 2, // ??
+            id: entity.id,
+            name: entity.name,
+            account_id: entity.account_id,
+            country,
+            city,
+        };
+
+        migrated_count += 1;
+
+        Some(new_entity)
+    });
+
+    info!(
+        " <<< Entity storage updated! Migrated {} Entities ✅",
+        migrated_count
+    );
+
+    // Return the weight consumed by the migration.
+    T::DbWeight::get().reads_writes(migrated_count as Weight + 1, migrated_count as Weight + 1)
+}
+
+fn migrate_nodes<T: Config>() -> frame_support::weights::Weight {
+    info!(" >>> Migrating nodes storage...");
+
+    let mut migrated_count = 0;
+
+    // We transform the storage values from the old into the new format.
+    Nodes::<T>::translate::<deprecated::Node<PubConfigOf<T>, InterfaceOf<T>>, _>(|k, node| {
+        info!("     Migrated node for {:?}...", k);
+
+        let location = match get_location::<T>(&node) {
+            Ok(loc) => loc,
+            Err(e) => {
+                info!(
+                    "location: [city: {:?}, country: {:?}, latitude: {:?}, longitude: {:?}]",
+                    core::str::from_utf8(&node.city).unwrap(),
+                    core::str::from_utf8(&node.country).unwrap(),
+                    core::str::from_utf8(&node.location.latitude).unwrap(),
+                    core::str::from_utf8(&node.location.longitude).unwrap()
+                );
+                info!("failed to parse location for node: {:?}, error: {:?}", k, e);
+                let default_location_input = LocationInput {
+                    city: BoundedVec::try_from(node::DEFAULT_CITY_NAME.to_vec()).unwrap(),
```

Comment:
Same, don't use unwrap. Either pour this functionality in a function that returns an error like here: 

https://github.com/threefoldtech/tfchain/blob/3c6640184cc2d4d291de037fb0e2f664fe26719b/substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs#L307


or handle every input seperatly. I prefer the former.

Reply:
Indeed better like this